lib/logstash/plugin_mixins/aws_config.rb

```diff
@@ -56,25 +59,21 @@ def aws_options_hash
       @logger.warn("Likely config error: Only one of access_key_id or secret_access_key was provided but not both.")
     end
 
-    if ((!@access_key_id || !@secret_access_key)) && @aws_credentials_file
-      access_creds = YAML.load_file(@aws_credentials_file)
-
-      @access_key_id = access_creds[:access_key_id]
-      @secret_access_key = access_creds[:secret_access_key]
-    end
-
-    opts = {}
-
-    if (@access_key_id && @secret_access_key)
-      opts[:access_key_id] = @access_key_id
-      opts[:secret_access_key] = @secret_access_key
+    if @access_key_id && @secret_access_key
+      opts = {
+        :access_key_id => @access_key_id,
+        :secret_access_key => @secret_access_key
+      }
+      opts[:session_token] = @session_token if @session_token
+    elsif @aws_credentials_file
+      opts = YAML.load_file(@aws_credentials_file)
+    else
+      raise LogStash::ConfigurationError.new('Neither credential file and credential was provided')
```

Comment:
If the ruby sdk cannot find the configuration in the environment it will fallback to IAM Role.

```
specified credentials -> environment -> ec metadata.
```